### PR TITLE
Issue #110 - Rollback ignores parameter overrides

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

**Review by merenbach — APPROVED** [review]

**Comment by jessesuen** [comment]
As discussed, we will create a new rollback RPC endpoint which has numeric identifiers of the previous app history. Rollback will be done using this interface. The CLI usage will work like:
```
argocd app rollback guestbook 34
```

**Comment by alexmt** [comment]
@jessesuen as we agreed rollback functionality is now available via separate API call. PTAL

File: server/application/application.go [hunk 1]
```diff
@@ -334,24 +334,69 @@ func (s *Server) PodLogs(q *PodLogsQuery, ws ApplicationService_PodLogsServer) e
...
+	}
+	var deploymentInfo *appv1.DeploymentInfo
+	for _, info := range app.Status.RecentDeployments {
+		if info.Id == rollbackReq.Id {
+			deploymentInfo = &info
+			break
+		}
+	}
+	if deploymentInfo == nil {
+		return nil, fmt.Errorf("application %s does not have deployment with id %v", rollbackReq.Name, rollbackReq.Id)
```
Comment (jessesuen) [inline server/application/application.go]: Instead of fmt.Errorf, can we use `status.Errorf(codes.InvalidArgument`

**Review by jessesuen — APPROVED** [review]

**Comment by jessesuen** [comment]
I will add `argo app history guestbook` and `argo app rollback guestbook <id>` after your changes

File: server/application/application.go (hunk 1 above)
Comment (alexmt) [reply server/application/application.go]: done. thank you!

**Pull request merged**